src/snowflake/cli/plugins/init/commands.py

```diff
@@ -0,0 +1,241 @@
+    if not template_metadata_path.exists():
+        raise FileNotFoundError(
+            f"Template does not have {TEMPLATE_METADATA_FILE_NAME} file"
+        )
+    with template_metadata_path.open(read_file_limit_mb=DEFAULT_SIZE_LIMIT_MB) as fd:
+        yaml_contents = yaml.safe_load(fd) or {}
+    return Template(template_root, **yaml_contents)
+
+
+def _remove_template_metadata_file(template_root: SecurePath) -> None:
+    (template_root / TEMPLATE_METADATA_FILE_NAME).unlink()
+
+
+def _prompt_for_value(variable: TemplateVariable, no_interactive: bool) -> Any:
+    if no_interactive:
+        if not variable.default:
+            raise ClickException(f"Cannot determine value of variable {variable.name}")
+        return variable.default
+
+    # override "unchecked type" with 'str', as Typer deduces type from the value of 'default'
+    type_ = variable.type.python_type if variable.type else str
+    prompt = variable.prompt if variable.prompt else variable.name
+    return typer.prompt(prompt, default=variable.default, type=type_)
+
+
+def _determine_variable_values(
+    variables_metadata: List[TemplateVariable],
+    variables_from_flags: Dict[str, Any],
+    no_interactive: bool,
+) -> Dict[str, Any]:
+    """
+    Prompt user for values not provided in [variables_from_flags].
+    If [no_interactive] is True, fill not provided variables with their default values.
+    """
+    result = {}
+
+    log.debug(
+        "Resolving values of variables: %s",
+        ", ".join(v.name for v in variables_metadata),
+    )
+    for variable in variables_metadata:
+        if variable.name not in variables_from_flags:
+            value = _prompt_for_value(variable, no_interactive)
+        else:
+            value = variables_from_flags[variable.name]
+            if variable.type:
+                value = variable.type.python_type(value)
+
+        result[variable.name] = value
+
+    return result
+
+
+def _validate_cli_version(required_version: str) -> None:
+    from packaging.version import parse
+    from snowflake.cli.__about__ import VERSION
+
+    if parse(required_version) > parse(VERSION):
+        raise ClickException(
+            f"Snowflake CLI version ({VERSION}) is too low - minimum version required"
+            f" by template is {required_version}. Please upgrade before continuing."
+        )
+
+
+@app.command(no_args_is_help=True)
+def init(
+    path: str = PathArgument,
+    template: Optional[str] = TemplateOption,
+    template_source: Optional[str] = SourceOption,
+    variables: Optional[List[str]] = VariablesOption,
+    no_interactive: bool = NoInteractiveOption,
+    **options,
+) -> CommandResult:
+    """
+    Creates project from template.
+    """
+    variables_from_flags = {
+        v.key: v.value for v in parse_key_value_variables(variables)
+    }
+    is_remote = any(
+        template_source.startswith(prefix) for prefix in ["git@", "http://", "https://"]  # type: ignore
```

Comment:
have we tested this with private git repo (with ssh keys set-up)?

Reply:
Yes, I've created private repository with dummy template - works as intended